ext/dba/dba.c

```diff
@@ -835,9 +844,12 @@ static void php_dba_open(INTERNAL_FUNCTION_PARAMETERS, bool persistent)
 		}
 	}
 
-	if (error || hptr->open(info, &error) != SUCCESS) {
+	if (error || hptr->open(info, &error) == FAILURE) {
 		dba_close(info);
-		php_error_docref(NULL, E_WARNING, "Driver initialization failed for handler: %s%s%s", hptr->name, error?": ":"", error?error:"");
+		if (EXPECTED(!EG(exception))) {
+			php_error_docref(NULL, E_WARNING, "Driver initialization failed for handler: %s%s%s",
+				hptr->name, error?": ":"", error?error:"");
```

Comment:
This is close to unreadable; I suggest to add spaces at least:
```suggestion
				hptr->name, error ? ": " : "", error ? error : "");
```